crates/core/src/rlp/decode.rs

```diff
@@ -313,6 +312,28 @@ impl<T1: RLPDecode, T2: RLPDecode, T3: RLPDecode> RLPDecode for (T1, T2, T3) {
     }
 }
 
+impl<T1: RLPDecode, T2: RLPDecode, T3: RLPDecode, T4: RLPDecode> RLPDecode for (T1, T2, T3, T4) {
```

Comment:
is this impl really needed?

Reply:
I needed it before we had the `Decoder` struct, but now it's unnecessary. Should I remove it?